### ISSUE TITLE
ERA001: False positive for tab indented code

### ISSUE THREAD

**Issue by sakshamgoyal-01** [astral-sh/ruff#25116]
### Summary

```py
# testing: Foo Bar Baz
```
Is being detected as code and throwing ERA001.

### Version

ruff 0.15.12

**Comment by ntBre** [comment]
Is this exact code emitting the diagnostic? I can only seem to reproduce the issue with something like:

```py
# testing: Foo
```

which resembles an annotation. The snippet in the issue shouldn't parse as valid Python code and thus won't trigger the rule. `testing` also isn't one of our code indicators, so this might be a bit different from #18461.

https://play.ruff.rs/ec810328-50d4-42c3-b8e0-c04ead111d13

**Comment by sakshamgoyal-01** [comment]
Difference between tabs and spaces. errors with tabs, no error with spaces

https://play.ruff.rs/95b9ac76-aed9-4880-b200-ee7220b586dc

**Comment by ntBre** [comment]
Ah, thank you! This seems unrelated to #18461 then. We should probably trim all Python whitespace instead of just spaces here:

https://github.com/astral-sh/ruff/blob/961f533be58616330f5e0ed7f7d632da1e1f8008/crates/ruff_linter/src/rules/eradicate/detection.rs#L78-L79
